### ISSUE TITLE
pyface/traitsui bug

### ISSUE THREAD

**Issue by zacsimile** [python-microscopy/python-microscopy#259]
Just did a fresh miniconda3 install on my Mac. Per our recent testing with cluster, I created a `python=3.6`, `numpy=1.4`, `wxpython=4.0.4` environment. I ran `visgui` and tried to open a file and received the following error. The strange thing is, this set of versions of `pyface`/`traits`/`traitsui` is listed as working in some of our other issues.

<details>
<summary>OSX packages</summary>
<pre>
# Name                    Version                   Build  Channel
alabaster                 0.7.12                   py36_0  
appnope                   0.1.0            py36hf537a9a_0  
attrs                     19.3.0                     py_0  
babel                     2.8.0                      py_0  
backcall                  0.1.0                    py36_0  
backports                 1.0                        py_2  
backports.lzma            0.0.14           py36hd0afa68_0  
beautifulsoup4            4.9.0                    py36_0  
blas                      1.0                         mkl  
bleach                    3.1.4                      py_0  
blosc                     1.15.0               hd9629dc_0  
bzip2                     1.0.8                h1de35cc_0  
ca-certificates           2020.1.1                      0  
cairo                     1.14.12              hc4e6be7_4  
certifi                   2020.4.5.1               py36_0  
cffi                      1.14.0           py36hc512035_1  
chardet                   3.0.4                 py36_1003  
cheroot                   8.3.0                    py36_0  
cherrypy                  18.6.0                   py36_0  
cloudpickle               1.4.1                      py_0  
conda                     4.8.3                    py36_0  
conda-build               3.18.11                  py36_0  
conda-package-handling    1.6.1            py36h1de35cc_0  
cryptography              2.9.2            py36ha12b0ac_0  
cycler                    0.10.0           py36hfc81398_0  
cython                    0.29.17          py36h0a44026_0  
cytoolz                   0.10.1           py36h1de35cc_0  
dask-core                 2.16.0                     py_0  
dbus                      1.13.14              h517e14e_0  
decorator                 4.4.2                      py_0  
defusedxml                0.6.0                      py_0  
dispatch                  1.0                      py36_0    david_baddeley
django                    1.11.8           py36hd476221_0  
docutils                  0.16                     py36_0  
entrypoints               0.3                      py36_0  
enum-compat               0.0.2            py36h14911fc_0    david_baddeley
expat                     2.2.6                h0a44026_0  
fftw                      3.3.4                         0    david_baddeley
filelock                  3.0.12                     py_0  
fontconfig                2.13.0               h5d5b041_1  
freetype                  2.9.1                hb4e5f40_0  
future                    0.18.2                   py36_0  
gettext                   0.19.8.1             hb0f4f8b_2  
glib                      2.63.1               h70d4741_1  
glob2                     0.7                        py_0  
hdf5                      1.10.4               hfa1e0ec_0  
icu                       58.2                 h0a44026_3  
idna                      2.9                        py_1  
imageio                   2.8.0                      py_0  
imagesize                 1.2.0                      py_0  
importlib_metadata        1.5.0                    py36_0  
importlib_resources       1.4.0                    py36_0  
intel-openmp              2019.4                      233  
ipykernel                 5.1.4            py36h39e3cac_0  
ipython                   7.13.0           py36h5ca1d4c_0  
ipython_genutils          0.2.0                    py36_0  
ipywidgets                7.5.1                      py_0  
jaraco.classes            3.1.0                      py_0  
jaraco.collections        3.0.0                      py_0  
jaraco.functools          3.0.1                      py_0  
jaraco.text               3.2.0                    py36_0  
jedi                      0.17.0                   py36_0  
jinja2                    2.11.2                     py_0  
joblib                    0.15.1                     py_0  
jpeg                      9b                   he5867d9_2  
jsonschema                3.2.0                    py36_0  
jupyter                   1.0.0                    py36_7  
jupyter_client            6.1.3                      py_0  
jupyter_console           6.1.0                      py_0  
jupyter_core              4.6.3                    py36_0  
kiwisolver                1.2.0            py36h04f5b5a_0  
libarchive                3.4.2                haa3ed63_0  
libcxx                    10.0.0                        1  
libedit                   3.1.20181209         hb402a30_0  
libffi                    3.3                  h0a44026_1  
libgfortran               3.0.1                h93005f0_2  
libiconv                  1.16                 h1de35cc_0  
liblief                   0.10.1               h0a44026_0  
libpng                    1.6.37               ha441bb4_0  
libsodium                 1.0.16               h3efe00b_0  
libtiff                   4.0.9                hcb84e12_2  
libxml2                   2.9.9                hf6e021a_1  
llvm-openmp               10.0.0               h28b9765_0  
lz4-c                     1.9.2                h0a44026_0  
lzo                       2.10                 h1de35cc_2  
markupsafe                1.1.1            py36h1de35cc_0  
matplotlib                3.1.3                    py36_0  
matplotlib-base           3.1.3            py36h9aa3819_0  
mistune                   0.8.4            py36h1de35cc_0  
mkl                       2019.4                      233  
mkl-service               2.3.0            py36hfbe908c_0  
mkl_fft                   1.0.15           py36h5e564d8_0  
mkl_random                1.1.0            py36ha771720_0  
more-itertools            8.3.0                      py_0  
mpld3                     0.3git           py36h1cdc3d8_0    david_baddeley
nbconvert                 5.6.1                    py36_0  
nbformat                  5.0.6                      py_0  
ncurses                   6.2                  h0a44026_1  
netifaces                 0.10.6           py36h455e006_0    david_baddeley
networkx                  2.4                        py_0  
notebook                  6.0.3                    py36_0  
numexpr                   2.7.1            py36hce01a72_0  
numpy                     1.14.6           py36h648b28d_5  
numpy-base                1.14.6           py36h6575580_5  
olefile                   0.46                     py36_0  
openssl                   1.1.1g               h1de35cc_0  
packaging                 20.3                       py_0  
pandas                    1.0.3            py36h6c726b0_0  
pandoc                    2.2.3.2                       0  
pandocfilters             1.4.2                    py36_1  
parso                     0.7.0                      py_0  
pcre                      8.43                 h0a44026_0  
pexpect                   4.8.0                    py36_0  
pickleshare               0.7.5                    py36_0  
pillow                    5.4.1            py36hb68e598_0  
pip                       20.0.2                   py36_3  
pixman                    0.38.0               h1de35cc_0  
pkginfo                   1.5.0.1                  py36_0  
portend                   2.5                        py_0  
prometheus_client         0.7.1                      py_0  
prompt-toolkit            3.0.4                      py_0  
prompt_toolkit            3.0.4                         0  
psutil                    5.7.0            py36h1de35cc_0  
ptyprocess                0.6.0                    py36_0  
py-lief                   0.10.1           py36haf313ee_0  
pycairo                   1.19.0           py36ha54c0a8_0  
pycosat                   0.6.3            py36h1de35cc_0  
pycparser                 2.20                       py_0  
pyface                    6.1.2                    py36_0  
pyfftw                    v0.11.1          py36hba1437f_0    david_baddeley
pygments                  2.6.1                      py_0  
pyme                      20.4.25                   dev_0    <develop>
pyme-depends              1.08                     py36_0    file:///Users/zachcm/miniconda3/envs/pyme/conda-bld
pymecompress              0.2.0                    py36_0    david_baddeley
pyopengl                  3.1.1a1                  py36_0  
pyopenssl                 19.1.0                   py36_0  
pyparsing                 2.4.7                      py_0  
pyqt                      5.9.2            py36h655552a_2  
pyrsistent                0.16.0           py36h1de35cc_0  
pysocks                   1.7.1                    py36_0  
pytables                  3.5.1            py36h5bccee9_0  
python                    3.6.10               hf48f09d_2  
python-dateutil           2.8.1                      py_0  
python-libarchive-c       2.9                        py_0  
python.app                2                       py36_10  
pytz                      2020.1                     py_0  
pywavelets                1.1.1            py36h1de35cc_0  
pyyaml                    5.3.1            py36h1de35cc_0  
pyzmq                     18.1.1           py36h0a44026_0  
qt                        5.9.7                h468cd18_1  
qtconsole                 4.7.4                      py_0  
qtpy                      1.9.0                      py_0  
readline                  8.0                  h1de35cc_0  
repoze.lru                0.7                      py36_0  
requests                  2.23.0                   py36_0  
ripgrep                   11.0.2               he32d670_0  
routes                    2.4.1                    py36_0  
ruamel_yaml               0.15.87          py36h1de35cc_0  
scikit-image              0.16.2           py36h6c726b0_0  
scikit-learn              0.22.1           py36h27c97d8_0  
scipy                     1.4.1            py36h9fa6033_0  
send2trash                1.5.0                    py36_0  
setuptools                46.4.0                   py36_0  
simplejson                3.17.0           py36h1de35cc_0  
sip                       4.19.8           py36h0a44026_0  
six                       1.14.0                   py36_0  
snappy                    1.1.7                he62c110_3  
snowballstemmer           2.0.0                      py_0  
soupsieve                 2.0.1                      py_0  
sphinx                    3.0.3                      py_0  
sphinxcontrib-applehelp   1.0.2                      py_0  
sphinxcontrib-devhelp     1.0.2                      py_0  
sphinxcontrib-htmlhelp    1.0.3                      py_0  
sphinxcontrib-jsmath      1.0.1                      py_0  
sphinxcontrib-qthelp      1.0.3                      py_0  
sphinxcontrib-serializinghtml 1.1.4                      py_0  
sqlite                    3.31.1               h5c1f38d_1  
tempora                   3.0.0                      py_0  
terminado                 0.8.3                    py36_0  
testpath                  0.4.4                      py_0  
tk                        8.6.8                ha441bb4_0  
toolz                     0.10.0                     py_0  
toposort                  1.1                pyh0993448_1    david_baddeley
tornado                   6.0.4            py36h1de35cc_1  
tqdm                      4.46.0                     py_0  
traitlets                 4.3.3                    py36_0  
traits                    6.0.0            py36h1de35cc_0  
traitsui                  6.1.3                      py_0  
ujson                     2.0.3            py36h0a44026_0  
urllib3                   1.25.8                   py36_0  
wcwidth                   0.1.9                      py_0  
webencodings              0.5.1                    py36_1  
wheel                     0.34.2                   py36_0  
widgetsnbextension        3.5.1                    py36_0  
wxpython                  4.0.4            py36hebc47a7_0  
xz                        5.2.5                h1de35cc_0  
yaml                      0.1.7                hc338f04_2  
zc.lockfile               2.0                        py_0  
zeroconf                  0.18.0                   pypi_0    pypi
zeromq                    4.3.1                h0a44026_3  
zipp                      3.1.0                      py_0  
zlib                      1.2.11               h1de35cc_3  
zstd                      1.4.4                h1990bb4_3  
</pre>
</details>

```
 Error whilst running File>&Open
==============================================
python-microscopy=20.04.25
python=3.6.10, platform=darwin
numpy=1.14.6, wx=4.0.4 osx-cocoa (phoenix) wxWidgets 3.0.5

Traceback
=========
Traceback (most recent call last):
  File "/Users/zachcm/Code/python-microscopy/PYME/ui/progress.py", line 107, in func
    return fcn(*args, **kwargs)
  File "/Users/zachcm/Code/python-microscopy/PYME/LMVis/visCore.py", line 402, in OnOpenFile
    self.OpenFile(filename)
  File "/Users/zachcm/Code/python-microscopy/PYME/LMVis/visCore.py", line 721, in OpenFile
    self.recipeView._layout()
  File "/Users/zachcm/Code/python-microscopy/PYME/recipes/vertical_recipe_display.py", line 122, in _layout
    pan = node.edit_traits(parent=item, kind=kind, view='pipeline_view_min')
  File "/Users/zachcm/miniconda3/envs/pyme/lib/python3.6/site-packages/traits/has_traits.py", line 1706, in edit_traits
    view = self.trait_view(view)
  File "/Users/zachcm/Code/python-microscopy/PYME/recipes/base.py", line 238, in trait_view
    import traitsui.api as tui
  File "/Users/zachcm/miniconda3/envs/pyme/lib/python3.6/site-packages/traitsui/api.py", line 36, in <module>
    from .editors.api import ArrayEditor
  File "/Users/zachcm/miniconda3/envs/pyme/lib/python3.6/site-packages/traitsui/editors/__init__.py", line 23, in <module>
    from .api import ArrayEditor
  File "/Users/zachcm/miniconda3/envs/pyme/lib/python3.6/site-packages/traitsui/editors/api.py", line 24, in <module>
    from .code_editor import CodeEditor
  File "/Users/zachcm/miniconda3/envs/pyme/lib/python3.6/site-packages/traitsui/editors/code_editor.py", line 37, in <module>
    class ToolkitEditorFactory(EditorFactory):
  File "/Users/zachcm/miniconda3/envs/pyme/lib/python3.6/site-packages/traitsui/editors/code_editor.py", line 49, in ToolkitEditorFactory
    mark_color = Color(0xECE9D8)
  File "/Users/zachcm/miniconda3/envs/pyme/lib/python3.6/site-packages/traits/trait_factory.py", line 40, in __call__
    return self.maker_function(*args, **metadata)
  File "/Users/zachcm/miniconda3/envs/pyme/lib/python3.6/site-packages/traits/traits.py", line 717, in Color
    return ColorTrait(*args, **metadata)
  File "/Users/zachcm/miniconda3/envs/pyme/lib/python3.6/site-packages/traitsui/toolkit_traits.py", line 8, in ColorTrait
    return toolkit().color_trait(*args, **traits)
  File "/Users/zachcm/miniconda3/envs/pyme/lib/python3.6/site-packages/traitsui/toolkit.py", line 109, in toolkit
    _toolkit = find_toolkit('traitsui.toolkits')
  File "/Users/zachcm/miniconda3/envs/pyme/lib/python3.6/site-packages/pyface/base_toolkit.py", line 263, in find_toolkit
    return import_toolkit(ETSConfig.toolkit, entry_point)
  File "/Users/zachcm/miniconda3/envs/pyme/lib/python3.6/site-packages/pyface/base_toolkit.py", line 227, in import_toolkit
    raise RuntimeError(msg)
RuntimeError: No traitsui.toolkits plugin could be loaded for wx
```

**Comment by David-Baddeley** [comment]
I suspect that the traitsui builds might be different on different platforms :(. This is going to be a right pain.

to summarise, the issue is `traitsui==6.1.3` on OSX. This is pointing more and more to at least needing to maintain our own packages for traitsui rather than relying on conda to do it properly.

**Comment by zacsimile** [comment]
Ah, man, I already went through this once in #92. Sorry. Downgrading to `pyface=6.0.0` and `traits=5.1.1` fixed it. As long as I do this, I can leave `traitsui=6.1.3` and everything works.

**Comment by zacsimile** [comment]
Maybe we should pin to these versions in `pyme-depends`?

**Comment by barentine** [comment]
I was able to leave pyface at 6.1.2, and traitsui at 6.1.3, but downgrading traits from 6.0.0 -> 5.1.1 as you suggested fixed things.

**Comment by zacsimile** [comment]
Fixed in python-microscopy/pyme-conda-recipes#5
